skimage/filters/_blur_effect.py

```diff
@@ -54,13 +54,17 @@ def blur_effect(image, h_size=11, channel_axis=None):
     n_axes = image.ndim
 
     if channel_axis is not None:
-        if not isinstance(channel_axis, int) or channel_axis >= image.ndim:
-            raise ValueError('channel_axis value is invalid')
-        else:
-            n_axes = n_axes - 1
+        try:
             # ensure color channels are in the final dimension
             image = np.moveaxis(image, channel_axis, -1)
-            image = rgb2gray(image)
+        except np.AxisError:
+            print('channel_axis must be one of the image array dimensions')
+            raise
+        except TypeError:
+            print('channel_axis must be an integer')
+            raise
```

Comment:
Thanks, this looks fine to me. I forget which version of NumPy first introduced `np.AxisError`, but I think it is earlier than our minimum supported version (indeed CI passed for the minimal dependencies case).

Reply:
Right, our minimum supported version for NumPy is `1.17.0` and `AxisError` was already available in `1.13.0`: https://numpy.org/doc/stable/release/1.13.0-notes.html?highlight=axiserror#compatibility-notes